eclipse-scout-core/src/code/Code.ts

```diff
@@ -117,16 +117,19 @@ export class Code<TCodeId> implements ObjectWithType {
 
   /**
    * Visits all children of this Code recursively without visiting this Code itself.
+   *
+   * By default, only active child codes are visited. To visit inactive child codes as well, set the `activeOnly` argument to false.
    */
-  visitChildren(visitor: TreeVisitor<Code<TCodeId>>): boolean | TreeVisitResult {
-    for (let i = 0; i < this.children.length; i++) {
-      let child = this.children[i];
+  visitChildren(visitor: TreeVisitor<Code<TCodeId>>, activeOnly = true): boolean | TreeVisitResult {
+    let children = this.children.filter(child => child.active || !activeOnly);
```

Comment:
Here, if activeOnly is false (=all codes should be visited), the filter could be skipped instead of filtering with all codes accepting the filter. But maybe this optimization is not necessary.